libs/header/p2p/metrics.go

```diff
@@ -0,0 +1,57 @@
+package p2p
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/metric/global"
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/metric/instrument/syncfloat64"
+)
+
+type metrics struct {
+	responseSize     syncfloat64.Histogram
+	responseDuration syncfloat64.Histogram
+}
+
+var (
+	meter = global.MeterProvider().Meter("header/p2p")
+)
+
+func (ex *Exchange[H]) RegisterMetrics() error {
```

Comment:
Instead, there should be a private field on Params that can be enabled by an option for Exchange.

Reply:
WithMetrics on nodebuilder will then supply this option to the FX and to Exchange

Reply:
As I mentioned in our discussion, I encountered FX issues with this approach, so we can afford to stay consistent to DASer ftm.